solidity/contracts/token/libs/MovableCollateralRouter.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {FungibleTokenRouter} from "./FungibleTokenRouter.sol";
+import {ValueTransferBridge} from "./ValueTransferBridge.sol";
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+abstract contract MovableCollateralRouter is AccessControl {
+    using SafeERC20 for IERC20;
+
+    constructor() {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    bytes32 public constant REBALANCER_ROLE = keccak256("REBALANCER_ROLE");
+
+    mapping(uint32 destinationDomain => bytes32 recipient)
+        public allowedDestinations;
+    mapping(uint32 destinationDomain => mapping(ValueTransferBridge bridge => bool isValidBridge))
+        public allowedBridges;
+
+    event CollateralMoved(
+        uint32 indexed domain,
+        bytes32 recipient,
+        uint256 amount,
+        address indexed rebalancer
+    );
+
+    error BadDestination(address rebalancer, uint32 domain);
+    error BadBridge(address rebalancer, ValueTransferBridge bridge);
+
+    function rebalance(
+        uint32 domain,
+        bytes32 recipient,
+        uint256 amount,
+        ValueTransferBridge bridge
+    ) external payable onlyRole(REBALANCER_ROLE) {
```

Comment:
should we/could we add a version that doesnt take a recipient? as I expect most initial usage patterns to do